test/Generator/EnumGeneratorTest.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace LaminasTest\Code\Generator;
+
+use Laminas\Code\Generator\EnumGenerator\EnumGenerator;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group Laminas_Code_Generator
+ * @group Laminas_Code_Generator_Php
+ */
+class EnumGeneratorTest extends TestCase
+{
+    /**
+     * @dataProvider validOptionSpecifications
+     * @psalm-param array{
+     *      name: string,
+     *      pureCases?: list<string>,
+     *      backedCases?: array{
+     *          type: string,
+     *          cases: array<string, int|string>
+     *      },
+     *      methods?: list<array{
+     *          name: string,
+     *          visibility?: string,
+     *          static?: bool,
+     *          returntype?: string,
+     *      }>
+     * } $options
+     */
+    public function testGenerateValidEnums(array $options, string $expected): void
+    {
+        self::assertSame($expected, EnumGenerator::withConfig($options)->generate());
+    }
+
+    /**
+     * @psalm-return iterable<string, array{
+     *     0: array{
+     *          name: string,
+     *          pureCases?: list<string>,
+     *          backedCases?: array{
+     *              type: string,
+     *              cases: array<string, int|string>
+     *          },
+     *          methods?: list<array{
+     *              name: string,
+     *              visibility?: string,
+     *              static?: bool,
+     *              returntype?: string,
+     *          }>
+     *      },
+     *      1: string
+     * }>
+     */
+    public function validOptionSpecifications(): iterable
+    {
+        yield 'pure enum without namespace' => [
+            [
+                'name'      => 'Suit',
+                'pureCases' => ['Hearts', 'Diamonds', 'Clubs', 'Spades'],
+            ],
+            <<<CODE
+enum Suit {
+    case Hearts;
+    case Diamonds;
+    case Clubs;
+    case Spades;
+}
+
+CODE,
```

Comment:
```suggestion
            <<<CODE
                enum Suit {
                    case Hearts;
                    case Diamonds;
                    case Clubs;
                    case Spades;
                }

                CODE,
```

Since PHP 7.3, heredocs and nowdocs support indentation; any whitespace preceding the end delimiter will be stripped from the start of lines in the document. It's a nice feature to make inline heredoc/nowdocs more readable

Reply:
Aah that's cool, it's definitely much better the way you're suggesting.